File: result/metagpt/claude/humaneval/78.c
#include <stdio.h>
#include <string.h>

int hex_key(const char* num) {
    const char* key = "2357BD";
    int out = 0;
    for (int i = 0; i < strlen(num); i++) {
        if (strchr(key, num[i]) != NULL) {
            out++;
        }
    }
    return out;
}

File: result/metagpt/deepseek/humaneval/78.rs
pub fn hex_key(num: &str) -> usize {
    const KEY: &str = "2357BD";
    num.chars().filter(|c| KEY.contains(*c)).count()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_hex_key() {
        assert_eq!(hex_key("AB2357"), 4);
        assert_eq!(hex_key("ABCDEF"), 1);
        assert_eq!(hex_key("123456"), 3);
    }
}